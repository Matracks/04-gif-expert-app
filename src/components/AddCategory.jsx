import propTypes from 'prop-types'
import { useState } from "react"


export const AddCategory = ( {onNewCategory} ) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ( {target} ) => {
        setInputValue( target.value )
    }

    const onSubmit = (event) => {
        event.preventDefault()

        const inputtrim = inputValue.trim()

        if ( inputtrim.length <= 1 ) {
            return
        }
        onNewCategory( inputtrim )
        setInputValue('')
    }

  return (
    <form onSubmit={ onSubmit } aria-label='form'>
        <input
            type='text'
            placeholder="Search Gif"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: propTypes.func.isRequired
  }