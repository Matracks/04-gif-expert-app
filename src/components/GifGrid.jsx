import { GifItem } from "./GifItem"
import { useFetchGift } from "../hooks/useFetchGift"


export const GifGrid = ( { category } ) => {

    const { images, isLoading } = useFetchGift( category )

    return (
        <>
            
            {
                isLoading
                ? (<h2>Cargando...</h2>)
                : <h2>{ category }</h2>
            }
            
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                            <GifItem
                                key={ image.id }
                                { ...image }
                            />     
                        ))
                }
            </div>
 
        </>
    )
}
