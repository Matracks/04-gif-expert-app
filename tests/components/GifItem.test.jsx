import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Tests in GifItem', () => {

  const title = 'Soy el titulo'
  const url = 'https://dragon-ball/otra.jpg'

  test('should match with the snapshot', () => { 

    const { container } = render( <GifItem title={title} url={url}/> )
    expect(container).toMatchSnapshot()

  })


  test('should be show a img with URL and ALT', () => { 

    render( <GifItem title={title} url={url}/> )
    // screen.debug()
    // expect( screen.getByRole('img').alt ).toBe( title )
    const { src, alt} = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( title )

   })


   test('should be show the title in the component', () => { 

    render( <GifItem title={title} url={url}/> )
    expect( screen.getByText( title )).toBeTruthy()

    })

})