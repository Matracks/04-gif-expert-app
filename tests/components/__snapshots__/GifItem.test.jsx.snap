// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy el titulo"
      src="https://dragon-ball/otra.jpg"
    />
    <p>
      Soy el titulo
    </p>
  </div>
</div>
`;
